### PR TITLE
Remove unused settings from main settings screen

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/6173

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [x] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: android/java/org/chromium/chrome/browser/preferences/BraveMainPreferences.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @Override
+    public void onResume() {
+        super.onResume();
+        updateBravePreferences();
+    }
+
+    private void updateBravePreferences() {
+        removePreferenceIfPresent(PREF_SIGN_IN);
+        removePreferenceIfPresent(PREF_ACCOUNT_SECTION);
+        removePreferenceIfPresent(PREF_DATA_REDUCTION);
```
Comment (simonhong) [inline android/java/org/chromium/chrome/browser/preferences/BraveMainPreferences.java]: @SergeyZhukovsky Should we remove data reduction and autofill assistant both?

File: android/java/org/chromium/chrome/browser/preferences/BraveMainPreferences.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+package org.chromium.chrome.browser.preferences;
+
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.v7.preference.Preference;
+
+import java.util.HashMap;
+
+// This exculdes some settings in main settings screen.
+public class BraveMainPreferences extends MainPreferences {
```
Comment (simonhong) [inline android/java/org/chromium/chrome/browser/preferences/BraveMainPreferences.java]: I'm thinking whether super class is better or subclass is better for customizing main preferences.

Comment (simonhong) [reply android/java/org/chromium/chrome/browser/preferences/BraveMainPreferences.java]: Changed to use super class. With this, we don't need to worry about the code that use `MainPreferences.class` directly.

File: patches/chrome-android-java-src-org-chromium-chrome-browser-signin-SigninManager.java.patch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/signin/SigninManager.java b/chrome/android/java/src/org/chromium/chrome/browser/signin/SigninManager.java
+index 1d63693d5979aabd516d8037e060b8ccf2f59c82..74751acd72886a894ba6e84c1e2e9e78c84fbb74 100644
+--- a/chrome/android/java/src/org/chromium/chrome/browser/signin/SigninManager.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/signin/SigninManager.java
+@@ -226,7 +226,7 @@ public class SigninManager implements AccountTrackerService.OnSystemAccountsSeed
+         assert nativeSigninManagerAndroid != 0;
+         assert delegate != null;
+         assert accountTrackerService != null;
+-        return new SigninManager(ContextUtils.getApplicationContext(), nativeSigninManagerAndroid,
++        return new BraveSigninManager(ContextUtils.getApplicationContext(), nativeSigninManagerAndroid,
```
Comment (simonhong) [inline patches/chrome-android-java-src-org-chromium-chrome-browser-signin-SigninManager.java.patch]: Trying to avoid this patch file.

Comment (simonhong) [reply patches/chrome-android-java-src-org-chromium-chrome-browser-signin-SigninManager.java.patch]: Removed this patch file by overriding `Java_SigninManager_create`.

File: android/java/org/chromium/chrome/browser/preferences/BraveMainPreferences.java (hunk 1 above)
Comment (SergeyZhukovsky) [reply android/java/org/chromium/chrome/browser/preferences/BraveMainPreferences.java]: I think so, as far as they are removed in the current version

**Comment by simonhong** [comment]
Merged because only audit was failed in CI.

**Pull request merged**